wherefore/KafkaConsumer.py

```diff
@@ -0,0 +1,215 @@
+            self._raise_from_kafka_exception(e)
+
+        r = []
+        for m in msgs:
+            r.append(Message(m.value(), m.timestamp()[1] / 1e3, m.offset()))
+
+        return r
+
+    def beginning_offset(self, topic: str, partition: int) -> int:
+        """
+        Get low offset for topic partition.
+
+        :param topic: Topic name.
+        :param partition: Partition.
+        :return: Low offset.
+        """
+        offsets = self._get_offsets(topic, partition)
+        return offsets[0]
+
+    def end_offset(self, topic: str, partition: int) -> int:
+        """
+        Get high offset for topic partition.
+
+        :param topic: Topic name.
+        :param partition: Partition.
+        :return: High offset.
+        """
+        offsets = self._get_offsets(topic, partition)
+        return offsets[1]
+
+    def _get_offsets(self, topic: str, partition: int):
+        tp = TopicPartition(topic, partition)
+
+        try:
+            offsets = self._consumer.get_watermark_offsets(tp)
+        except KafkaException as e:
+            self._raise_from_kafka_exception(e)
+
+        return offsets
+
+    def topics(self) -> Set[str]:
+        """
+        Get topic names.
+
+        :return: Set of topic names.
+        """
+        try:
+            md = self._consumer.list_topics()
+        except KafkaException as e:
+            self._raise_from_kafka_exception(e)
+
+        topics = set()
+        for k in md.topics.keys():
+            topics.add(k)
+
+        return topics
+
+    def partitions_for_topic(self, topic: str) -> Set[int]:
+        """
+        Get partitions for topic.
+
+        :param topic: Topic name.
+        :return: Set of partitions for topic.
+        """
+        try:
+            md = self._consumer.list_topics()
+        except KafkaException as e:
+            self._raise_from_kafka_exception(e)
+
+        try:
+            tmd = md.topics[topic]
+        except KeyError:
+            return None
+
+        partitions = set()
+        for k in tmd.partitions.keys():
+            partitions.add(k)
+
+        return partitions
+
+    def assign(self, topic: str, partitions: List[int]):
+        """
+        Assign topic and partitions to consumer.
+
+        :param topic: Topic name.
+        :param partitions: List of partitions.
+        """
+        tps = [TopicPartition(topic, p) for p in partitions]
+
+        try:
+            self._consumer.assign(tps)
+        except KafkaException as e:
+            self._raise_from_kafka_exception(e)
+
+    def offsets_for_times(self, partitions_and_times: Dict[int, int]) -> Dict[int, int]:
+        """
+        Get offsets per partition from timestamps in ms for assigned topic.
+
+        :param partitions_and_times: Dict with partitions and timestamps in ms.
+        :return: Dict with partitions and offsets (None if timestamp is after last message).
+        """
+        try:
+            a = self._consumer.assignment()
```

Comment:
I don't think we should be using single character names for things which have wide scope
```suggestion
            assigned_partitions = self._consumer.assignment()
```